lib/wallet/src/Cardano/Wallet/Primitive/Slotting.hs

```diff
@@ -401,6 +406,14 @@ getSystemStart :: TimeInterpreter m -> SystemStart
 getSystemStart TimeInterpreter{blockchainStartTime} =
     let (StartTime t) = blockchainStartTime in SystemStart t
 
+toTimeTranslation
+    :: TimeInterpreter (ExceptT PastHorizonException IO)
+    -> IO TimeTranslation
+toTimeTranslation timeInterpreter = do
+    info <- runExceptT (toEpochInfo timeInterpreter)
+        >>= either throwIO (pure . hoistEpochInfo runExcept)
```

Comment:
In practice, we should never hit this `throwIO` per the removed comment from `Shelley.Transaction` Possibly we could still hint at this being the case.